code/game/objects/structures/door_assembly.dm

```diff
@@ -50,8 +50,8 @@
 	if(doorname)
 		. += span_notice("There is a small <i>paper</i> placard on the assembly labelled \"[doorname]\".")
 
-/obj/structure/door_assembly/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/pen))
+/obj/structure/door_assembly/attackby(obj/item/W, mob/living/user, params)
+	if(istype(W, /obj/item/pen) && (!istype(user) || !user.combat_mode))
```

Comment:
user should always be mob/living, so you can remove the istype check

Reply:
```suggestion
/obj/structure/door_assembly/attackby(obj/item/W, mob/living/user, params)
	if(istype(W, /obj/item/pen) && !user.combat_mode)
```